realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -462,6 +462,25 @@ public void setEmbedded(boolean embedded) {
         }
     }
 
+    /**
+     * Checks whether a given class property can represent a valid embedded object
```

Comment:
This Javadoc doesn't seem entirely accurate. It will return the linked type regardless of classes being embedded or not.

Reply:
I think it makes sense. I will change it to return the class name regardless of which object it is called upon and then move the assertion for a valid type to the `createEmbeddedObject` method. But then I don't think this method needs to be made public, as it's used only internally. I will make it package protected instead.